data_model/src/transaction.rs

```diff
@@ -329,7 +339,6 @@ mod candidate {
     impl SignedTransactionCandidate {
         fn validate(self) -> Result<SignedTransactionV1, &'static str> {
             self.validate_instructions()?;
-            self.validate_signature()?;
```

Comment:
I don't like this. Transactions with incorrect signatures might creep into Iroha. I'd rather that you make another `SignedTransactionCandidate` in `data_model::block` which will have different implementation, i.e. it will not verify genesis signatures

Reply:
I agree, but what to do with `validate_transaction` in executor? It will deserialize `SignedTransaction` at WASM side, so it will fail for genesis transaction if we keep validate signature during deserialization

Reply:
> One option I can think of is to make `SignedTransaction` validation `#[cfg(not(target_family = "wasm"))]`

I like this suggestion. Can we say that since data model entities are always provided by the host there is no need to validate them on the wasm side? If so, we can can open a ticket to generalize this approach to all entities, although it seems kinda hacky and I'm not sure if this won't bite us in the ass

Reply:
Another possibility is to hardcode some public key as genesis and say that every Iroha deployment will have this as genesis account_id. In that case we can remove `account` from `genesis.json`. This is more or less how we had it except that `kagami` wouldn't have to receive private key but it already knows it's value. @Erigara @s8sato ?

Reply:
let's discuss tomorrow on standup

Reply:
I was thinking about such opportunity and for example pick a key with all 0s, however as turned out such public key is not secure in `Ed25519`, because it's kinda identity element.

So if we going to pick some hardcoded public key users might have some trust issues if we picked it on purpose to have some kind of backdoor.

Alternative might be to extend `AccountId` to either be public key or arbitrary address which doesn't have corresponding private key and can't be used to sign transactions. It was [suggested](https://github.com/hyperledger/iroha/issues/2085#issuecomment-1980557101) by @s8sato for multisig accounts.

Reply:
We can use the same multihash format  but with [`identity`](https://github.com/multiformats/multicodec/blob/8fc4918c9e7c6cb8f1d0501523d5a0943c46a157/table.csv#L2) code

Reply:
After discussion it was noted that we still want genesis to sign transaction/block in which case my proposal is not relevant.

Reply:
The complexity seems to come from trying to strip away the functionality of digital signatures while taking advantage of the transaction processing flow.
There might not have been any problem in the first place: if the privilege comes from block height 0, what is the actual harm even if the genesis account with the correct private key appears after genesis?

Reply:
> what is the actual harm even if the genesis account with the correct private key appears after genesis?

I like this point. Especially if someone has misconfigured the blockchain in genesis.json, for example they haven't transfered ownership

Reply:
Yeah, currently custom genesis can grant any privileges to any account. But they agreed anyway. They joined the network while still being able to see that genesis block